prometheus_client/exposition.py

```diff
@@ -343,8 +357,13 @@ def handle():
 
 
 def push_to_gateway(
-        gateway, job, registry, grouping_key=None, timeout=30,
-        handler=default_handler):
+    gateway: str,
+    job: str,
+    registry: CollectorRegistry,
+    grouping_key: Optional[Dict[str, str]] = None,
```

Comment:
if grouping key is reused below, if it has a constant structure, may want to use a `TypedDict`

Reply:
I don't think it it has a constant structure, and `TypedDict` isn't available earlier than 3.8 anyway :(. I really want to use `Protocol` in a few places as well but cannot yet.

Reply:
It looks like you can use typing-extensions for `TypedDict` and `Protocol`.